core/src/fields.rs

```diff
@@ -705,46 +705,85 @@ impl MongoStatement for MongoFields {
             8 => Bson::Int32({
-                let l = get_meta_data()?.octet_length;
+                let l = get_meta_data()?.transfer_octet_length;
                 match l {
                     None => definitions::SQL_NO_TOTAL as i32,
                     Some(l) => l as i32,
                 }
             }),
-            9 => Bson::Int32(get_meta_data()?.scale.unwrap_or(0) as i32),
+            // DECIMAL_DIGITS
+            9 => match get_meta_data()?.decimal_digits {
+                // NULL is returned for data types where DECIMAL_DIGITS is not applicable.
+                None => Bson::Null,
+                Some(dec_dg) => Bson::Int32(dec_dg as i32),
+            },
+            // NUM_PREC_RADIX
             10 => match get_meta_data()?.sql_type {
+                // For numeric data types. 10 means that the values in COLUMN_SIZE and DECIMAL_DIGITS
+                // give the maximum number of digits and maximum number of digits to the right of the
+                // decimal point allowed for the column respectively.
                 SqlDataType::SQL_INTEGER | SqlDataType::SQL_DOUBLE | SqlDataType::SQL_DECIMAL => {
                     Bson::Int32(10)
                 }
                 _ => Bson::Null,
             },
+            // NULLABLE
             11 => Bson::Int32(get_meta_data()?.nullability as i32),
+            // REMARKS
             12 => Bson::String("".to_string()),
+            // COLUMN_DEF
             13 => Bson::Null,
+            // SQL_DATA_TYPE
             14 => Bson::Int32(get_meta_data()?.non_concise_type as i32),
+            // SQL_DATETIME_SUB
             15 => match get_meta_data()?.sql_code {
                 None => Bson::Null,
                 Some(x) => Bson::Int32(x as i32),
             },
-            16 => Bson::Int32({
-                let l = get_meta_data()?.octet_length;
-                match l {
-                    None => definitions::SQL_NO_TOTAL as i32,
-                    Some(_) => 0i32,
-                }
-            }),
+            // CHAR_OCTET_LENGTH
+            // The maximum length in bytes of a character or binary data type column.
+            // For all other data types, this column returns a NULL.
+            16 => match get_meta_data()?.sql_type {
+                // For numeric data types. 10 means that the values in COLUMN_SIZE and DECIMAL_DIGITS
+                // give the maximum number of digits and maximum number of digits to the right of the
+                // decimal point allowed for the column respectively.
+                SqlDataType::SQL_VARCHAR
+                | SqlDataType::SQL_WVARCHAR
+                | SqlDataType::SQL_VARBINARY => match get_meta_data()?.char_octet_length {
+                    None => Bson::Int32(definitions::SQL_NO_TOTAL as i32),
+                    Some(char_octet_length) => Bson::Int32(char_octet_length as i32),
+                },
+                _ => Bson::Null,
+            },
```

Comment:
```suggestion
            16 => {
                let meta_data = get_meta_data()?;
                // For numeric data types. 10 means that the values in COLUMN_SIZE and DECIMAL_DIGITS
                // give the maximum number of digits and maximum number of digits to the right of the
                // decimal point allowed for the column respectively.
                if matches!(
                    meta_data.sql_type,
                    SqlDataType::SQL_VARCHAR
                        | SqlDataType::SQL_WVARCHAR
                        | SqlDataType::SQL_VARBINARY
                ) {
                    Bson::Int32(
                        meta_data
                            .char_octet_length
                            .unwrap_or(definitions::SQL_NO_TOTAL) as i32,
                    )
                } else {
                    Bson::Null
                }
            }
```

Reply:
It might actually make sense to add an `is_char_type()` to `SqlDataType` that has that `matches!` in it, as well.

Reply:
Yes, I was thinking of doing that too.
There is only that one place where I needed this kind of rule, so I didn't make that change.
The other places, I baked the rule in the value itself.
But I think it might be valuable to have helpers for is_numeric, is_character, is_binary, is_datetime, etc...
I'll create a quickwin JIRA for this.